src/main/java/com/marklogic/client/helper/DatabaseClientConfig.java

```diff
@@ -4,7 +4,11 @@
 
 import com.marklogic.client.DatabaseClientFactory.Authentication;
 import com.marklogic.client.DatabaseClientFactory.SSLHostnameVerifier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
 
+@Component
```

Comment:
This makes me a little nervous. Generally, I think of Component being used for application-specific classes. 

How about subclassing DatabaseClientConfig with a Spring-specific subclass so that this class can remain unaware of Spring? i.e. SpringDatabaseClientConfig